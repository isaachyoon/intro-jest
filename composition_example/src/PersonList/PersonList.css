.container {
  width: 100%;
  padding: 1rem;
  border: solid 1px black;
  display: flex;
  justify-content: space-around;
  align-items: center;
}