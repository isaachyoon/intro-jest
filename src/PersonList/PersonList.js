import React from 'react';
import Person from '../Person/Person';
import './PersonList.css';

function PersonList() {
  const peopleData = [
      { id: '001', name: 'Rich'},
      { id: '002', name: 'Joe' },
      { id: '003', name: 'Chris'}, 
      { id: '004', name: 'Tauhida' },
      { id: '005', name: 'Elias' }, 
      { id: '006', name: 'Rita' }
  ];

  const people = peopleData.map((person, i) =>
      <Person key={i} />
  );
  
  return (
    <div className="personList">
      <h3>Person List</h3>
      <div className="container">
        {people}
      </div>
    </div>
  );
};

export default PersonList;