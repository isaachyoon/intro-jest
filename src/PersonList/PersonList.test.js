import React from 'react';
import ReactDOM from 'react-dom';
import PersonList from './PersonList';

import renderer from 'react-test-renderer';

describe('<PersonList />', () => {
  it('PersonList mounts', ()=> {
    const div = document.createElement('div');
    ReactDOM.render(<PersonList/>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  const obj = {id: '001', name: 'Rich'};
  it('matches snapshot', () => {
    const snap = renderer.create(<PersonList data={obj}/>).toJSON();
    expect(snap).toMatchSnapshot();
  });
})