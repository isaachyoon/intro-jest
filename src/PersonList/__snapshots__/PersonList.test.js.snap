// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonList /> matches snapshot 1`] = `
<div
  className="personList"
>
  <h3>
    Person List
  </h3>
  <div
    className="container"
  >
    <div
      className="person"
    >
      <p>
        001
      </p>
      <p>
        Rich
      </p>
    </div>
    <div
      className="person"
    >
      <p>
        002
      </p>
      <p>
        Joe
      </p>
    </div>
    <div
      className="person"
    >
      <p>
        003
      </p>
      <p>
        Chris
      </p>
    </div>
    <div
      className="person"
    >
      <p>
        004
      </p>
      <p>
        Tauhida
      </p>
    </div>
    <div
      className="person"
    >
      <p>
        005
      </p>
      <p>
        Elias
      </p>
    </div>
    <div
      className="person"
    >
      <p>
        006
      </p>
      <p>
        Rita
      </p>
    </div>
  </div>
  <div>
    hi
  </div>
</div>
`;
